### PR TITLE
Process created by ForkingTestClassProcessor should have native services disabled

### PR DESCRIPTION
**Pull request by asodja**
But not via `-Dorg.gradle.native=false` flag.

Fixes https://github.com/gradle/gradle/issues/28401

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
I don't think ObjectiveC test failures are related to that, since they happen also on other branches, see https://gradle.slack.com/archives/CBSJ2GUTV/p1710171682036509

File: subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java [hunk 1]
```diff
@@ -59,6 +59,10 @@ public interface WorkerProcessBuilder extends WorkerProcessSettings {
 
     void enableJvmMemoryInfoPublishing(boolean shouldPublish);
 
+    void useNativeServices(Boolean useNativeServices);
```
Comment (jvandort) [inline subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java:62]: Should be marked `@Nullable`. Same with the method that implements this

**Review by jvandort — APPROVED** [review]

**Review by wolfs — APPROVED** [review]

File: subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java:62]: I think a much better solution would be to use an enum with three values (`ENABLE`, `DEFAULT`, `DISABLE`), like we do in many places, e.g. with `WatchMode`:

https://github.com/gradle/gradle/blob/b8190353a5a319cb1a2af993bc76feb13f670bf7/platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/WatchMode.java#L22-L40

Comment (jvandort) [reply subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java:62]: Also works for me

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Pull request merged**